rfcs/react-components/convergence/slot-children-render-function.md

```diff
@@ -0,0 +1,650 @@
+              )}
+            </slots.expandIcon>
+          )}
+          {slots.icon && <slots.icon {...slotProps.icon} />}
+          {slotProps.root.children}
+          {state.expandIconPosition === 'end' && (
+            <slots.expandIcon {...slotProps.expandIcon}>
+              {resolveChildren(
+                slotProps.expandIcon,
+                <ChevronRightRegular style={{ transform: `rotate(${expandIconRotation}deg)` }} />,
+              )}
+            </slots.expandIcon>
+          )}
+        </>,
+      )}
+    </slots.button>,
+  )}
+</slots.root>
+```
+
+#### Pros and Cons
+
+##### Pros
+
+1. It would solve the problem?!
+
+##### Cons
+
+1. Render methods will become more complex
+
+### Option C: Custom JSX Pragma without internal changes
+
+A minimal version of option A, but without introducing internal changes. If we focus on the problem itself,
+slot children render function, the main requirement would be to maintain `defaultProps.children` until the last possible
+moment so that it can be introduced once again back to a children render function.
+
+This proposal would take advantage of `Symbol` to introduce a "private" method to the result of `resolveShorthand`,
+to maintain the value of `defaultProps.children`. That value would then later on be consumed by a minimal custom JSX pragma
+that on the specific case of children render function this pragma would consume this `defaultProps.children` to ensure proper functionality.
+
+```diff
+const defaultPropsChildrenSymbol = Symbol('fuiSlotDefaultPropsChildren')
+
+export const resolveShorthand: ResolveShorthandFunction = (value, options) => {
+  //...
+
+  return defaultProps ? {
+    ...defaultProps,
+    ...resolvedShorthand,
++   [defaultPropsChildrenSymbol]: defaultProps?.children
+  } : resolvedShorthand;
+};
+```
+
+https://codesandbox.io/s/wispy-leftpad-f8yi37?file=/example.tsx.
+
+#### Pros and Cons
+
+##### Pros
+
+1. No internal changes required
+2. No migration is required
+
+##### Cons
+
+1. Requires extra build step
+
+### Option D: Option A + Option C
+
+Taking the strategy provided on Option C to hide required data with `Symbol` inside the result value of `resolveShorthand`
+and applying it to not only move forward the `defaultProps.children` (from Option C) but also `componentType` (from Option A)
+we can completely get rid of `getSlots` invocation, as all the data required to render a slot will be provided by `resolveShorthand` and only consumed at render by a custom pragma.
+
+```diff
+const defaultPropsChildrenSymbol = Symbol('fuiSlotDefaultPropsChildren')
+const componentTypeSymbol = Symbol('fuiSlotComponentType')
+
+export const resolveShorthand: ResolveShorthandFunction = (value, options) => {
+  //...
+
+  return defaultProps ? {
+    ...defaultProps,
+    ...resolvedShorthand,
++   [defaultPropsChildrenSymbol]: defaultProps?.children
++   [componentTypeSymbol]: options.componentType
```

Comment:
I would rather make an explicit argument (probably `first`), so the contract will be clearer and we will avoid questions like "why it renders a div?!"